sdk/core/azure-core/test/ut/exception_test.cpp

```diff
@@ -98,3 +100,55 @@ TEST(RequestFailedException, EmptyValues)
                   Azure::Core::Http::HttpStatusCode::None))),
       std::string::npos);
 }
+
+TEST(RequestFailedException, Constructible)
+{
+  EXPECT_FALSE((ClassTraits<RequestFailedException>::is_constructible()));
+  EXPECT_FALSE((ClassTraits<RequestFailedException>::is_trivially_constructible()));
+  EXPECT_FALSE((ClassTraits<RequestFailedException>::is_nothrow_constructible()));
+  EXPECT_FALSE(ClassTraits<RequestFailedException>::is_default_constructible());
+  EXPECT_FALSE(ClassTraits<RequestFailedException>::is_trivially_default_constructible());
+  EXPECT_FALSE(ClassTraits<RequestFailedException>::is_nothrow_default_constructible());
+}
+
+TEST(RequestFailedException, Destructible)
+{
+  EXPECT_TRUE(ClassTraits<RequestFailedException>::is_destructible());
+  EXPECT_FALSE(ClassTraits<RequestFailedException>::is_trivially_destructible());
```

Comment:
@antkmsft, @LarryOsterman I am seeing a bunch of true/false differences for `is_destructible` vs `is_trivially_destructible`, or `is_copy_constructible` vs `is_trivially_copy_constructible`. 

Before we codify the current behavior in tests, what's our guidance for types we have in Core that clarify when these type traits should be true vs false?

We have the 0-3-5 rule, how do I know, when looking at these tests, that we are conforming to that guideline?
https://azure.github.io/azure-sdk/cpp_implementation.html#cpp-design-logical-rule-of-zero

```text
✅DO In class types, implement the “rule of zero”, the “rule of 3”, or the “rule of 5”. That is, of the special member functions, a type should implement exactly one of the following:
- No copy constructor, no copy assignment operator, no move constructor, no move assignment operator, or destructor.
- A copy constructor, a copy assignment operator, no move constructor, no move assignment operator, and a destructor.
- A copy constructor, a copy assignment operator, a move constructor, a move assignment operator, and a destructor.
```

Reply:
@CaseyCarter in case you have thoughts on how we make sure these tests match what we intend the behavior to be.

Reply:
These tests seem to be validating that no observable property of any of our types changes, which seems like overkill in many cases. Do we document, for example, that `RequestFailedException` is not trivially copy constructible? Do our customers care? Do we expect there to be a hue and cry if it becomes trivially copy assignable during maintenance?

Reply:
For me, the concern is the invert - when something which is trivially copy assignable ceases to be. That is a breaking change that should be called out in release notes.

If we have tests that verify these invariants, they act as canaries - changing something to become trivially copy constructable isn't a problem, but by having these tests, we'll see the change because the tests are going to change.